dd-sdk-android/src/test/kotlin/com/datadog/android/log/internal/logger/LogcatLogHandlerTest.kt

```diff
@@ -84,4 +84,52 @@ internal class LogcatLogHandlerTest {
                     "\$runnable\$1"
             )
     }
+
+    @Test
+    fun `resolves valid stack trace element when wrapped in timber`() {
+
+        // Given
+        val rawStackTrace =
```

Comment:
since I'm creating fake stack trace which is not bound to the real runtime execution, it seems there is no need to create a similar test in the `javaTest` folder. please let me know if I'm wrong.

Reply:
Indeed that sounds unnecessary

Reply:
I understood that @xgouchet will introduce you later to Elmyr :). Having this in mind you could re - write this test to introduce more randomness as :
```
    @Test
    fun `resolves valid stack trace element when wrapped in timber`(forge: Forge) {

        // Given
        val ignoredClassNamesRegEx = LogcatLogHandler.ignoredClassNames.joinToString("|")
            .replace(".", "\\.")
        val expectedElementIndex = forge.anInt(min=2,max = 10)
        val stackTrace = Array<StackTraceElement>(
            expectedElementIndex +
                forge.anInt(min=1,max = 10)
        ) {
            if (it < expectedElementIndex / 2) {
                StackTraceElement(
                    forge.aStringMatching("((com\\.datadog\\.android)|(timber\\.log))(.+)"),
                    forge.anAlphabeticalString(),
                    forge.anAlphabeticalString(),
                    it
                )
            } else if (it < expectedElementIndex) {
                StackTraceElement(
                    forge.aStringMatching("($ignoredClassNamesRegEx)"),
                    forge.anAlphabeticalString(),
                    forge.anAlphabeticalString(),
                    it
                )
            } else {
                StackTraceElement(
                    forge.aStringMatching("[^ctCT](.+)($ignoredClassNamesRegEx)[a]"),
                    forge.anAlphabeticalString(),
                    forge.anAlphabeticalString(),
                    it
                )
            }
        }

        // When
        val element = testedHandler.findValidCallStackElement(stackTrace)

        // Then
        checkNotNull(element)
        assertThat(element)
            .isEqualTo(stackTrace[expectedElementIndex])
    }
```